core/common/iterator/src/main/java/org/eclipse/rdf4j/common/iteration/AbstractCloseableIteration.java

```diff
@@ -57,8 +57,8 @@ public final void close() throws X {
 	 * Called by {@link #close} when it is called for the first time. This method is only called once on each iteration.
 	 * By default, this method does nothing.
 	 *
-	 * @throws X
+	 *
 	 */
-	protected void handleClose() throws X {
+	protected void handleClose() {
 	}
```

Comment:
One issue I've seen when profiling is that the `handleClose()` method is overridden multiple times in a class hierarchy. This reduces the performance because the JVM struggles with knowing which method to actually call and needs to use a method stub per object before it can optimize the call site. 

So I would like to duplicate up the code instead of having it extend this class.

Reply:
Do we even still need this abstract class, really? Given that the only thing it really adds is a private boolean flag, its reuse value is very limited.